### PR TITLE
Add decorator

### PR DESCRIPTION
**Pull request by SergSkol**
-Add Nameable class as parent for Person
-Add Decorator class and his childs

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
```
Comment (zieeco) [inline decorator.rb:3]: - Hey, let's kindly add the method to access the nameable to give the read/write access to the name. 
`attr_accessor :nameable`

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  def initialize(nameable)
+    super()
```
Comment (zieeco) [inline decorator.rb:5]: - Let's also remove the super as this is currently not needed.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @SergSkol ,


Great job so far!  👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice class decoration. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (SergSkol) [reply decorator.rb:3]: OK, it's done! thanks!

File: decorator.rb (hunk 2 above)
Comment (SergSkol) [reply decorator.rb:5]: OK, it's done! thanks!

Comment (SergSkol) [reply decorator.rb:5]: But now I recieved the following offense from rubocop:

decorator.rb:4:3: W: Lint/MissingSuper: Call super to initialize state of the parent class.
  def initialize(nameable) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^

**Comment by SergSkol** [comment]
> 

Code reviewer "zieeco" has required to remove super():  "Let's also remove the super as this is currently not needed.". 
I did it and recieved linters error:   decorator.rb:4:3: W: Lint/MissingSuper: Call super to initialize state of the parent class.

Could you please advise what should I do: add super, or remove?

**Comment by MaganaAsiati** [comment]
> > 
> 
> Code reviewer "zieeco" has required to remove super(): "Let's also remove the super as this is currently not needed.". I did it and recieved linters error: decorator.rb:4:3: W: Lint/MissingSuper: Call super to initialize state of the parent class.
> 
> Could you please advise what should I do: add super, or remove?

Please bring it back. you need it. you you did inheritance from child to parent class

**Comment by SergSkol** [comment]
> > > 
> > 
> > 
> > Code reviewer "zieeco" has required to remove super(): "Let's also remove the super as this is currently not needed.". I did it and recieved linters error: decorator.rb:4:3: W: Lint/MissingSuper: Call super to initialize state of the parent class.
> > Could you please advise what should I do: add super, or remove?
> 
> Please bring it back. you need it. you you did inheritance from child to parent class

Thank you! I did return it back.

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @SergSkol , I'm @MaganaAsiati , your reviewer for this PR  :wave: 
Great work on making the changes requested by a previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 





## Highlights
- Descriptive pull request :heavy_check_mark:👨🏻‍💻
- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] Check the comments under the review.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
Cheers and Happy coding! 🤓 👏👏👏

**Pull request merged**